netty/src/test/java/io/grpc/netty/ProtocolNegotiatorsTest.java

```diff
@@ -383,53 +515,148 @@ public void httpProxy_500() throws Exception {
+              ctx.writeAndFlush(res);
+              ctx.close();
+            }
+          }
+        });
+      }
+    };
     Channel serverChannel = new ServerBootstrap().group(elg).channel(LocalServerChannel.class)
-        .childHandler(mockHandler)
+        .childHandler(childHandler)
         .bind(proxy).sync().channel();
 
     ProtocolNegotiator nego =
         ProtocolNegotiators.httpProxy(proxy, null, null, ProtocolNegotiators.plaintext());
-    ChannelHandler handler = nego.newHandler(grpcHandler);
+    ChannelHandler handler = nego.newHandler(FakeGrpcHttp2ConnectionHandler.noopHandler());
     Channel channel = new Bootstrap().group(elg).channel(LocalChannel.class).handler(handler)
         .register().sync().channel();
     pipeline = channel.pipeline();
     // Wait for initialization to complete
     channel.eventLoop().submit(NOOP_RUNNABLE).sync();
-    // The grpcHandler must be in the pipeline, but we don't actually want it during our test
-    // because it will consume all events since it is a mock. We only use it because it is required
-    // to construct the Handler.
-    pipeline.remove(grpcHandler);
+    ChannelFuture wf =
+        channel.writeAndFlush(ByteBufUtil.writeUtf8(channel.alloc(), "isThisThingOn"));
     channel.connect(host).sync();
     serverChannel.close();
-    ArgumentCaptor<ChannelHandlerContext> contextCaptor =
-        ArgumentCaptor.forClass(ChannelHandlerContext.class);
-    Mockito.verify(mockHandler).channelActive(contextCaptor.capture());
-    ChannelHandlerContext serverContext = contextCaptor.getValue();
 
-    final String golden = "isThisThingOn?";
-    ChannelFuture negotiationFuture = channel.writeAndFlush(bb(golden, channel));
+    Object first = serverReads.poll(5, TimeUnit.SECONDS);
+    assertThat(first).isInstanceOf(HttpRequest.class);
+    HttpRequest req = (HttpRequest) first;
 
-    // Wait for sending initial request to complete
-    channel.eventLoop().submit(NOOP_RUNNABLE).sync();
-    ArgumentCaptor<Object> objectCaptor = ArgumentCaptor.forClass(Object.class);
-    Mockito.verify(mockHandler)
-        .channelRead(any(ChannelHandlerContext.class), objectCaptor.capture());
-    ByteBuf request = (ByteBuf) objectCaptor.getValue();
-    request.release();
-
-    assertFalse(negotiationFuture.isDone());
-    String response = "HTTP/1.1 500 OMG\r\nContent-Length: 4\r\n\r\noops";
-    serverContext.writeAndFlush(bb(response, serverContext.channel())).sync();
-    thrown.expect(ProxyConnectException.class);
-    try {
-      negotiationFuture.sync();
-    } finally {
-      channel.close();
+    assertThat(req.method()).isEqualTo(HttpMethod.CONNECT);
+    assertThat(req.uri()).isEqualTo("specialHost:314");
+    assertThat(req.headers().get(HttpHeaderNames.HOST)).isEqualTo("specialHost:314");
+
+    Object second = serverReads.poll(5, TimeUnit.SECONDS);
+    assertThat(second).isInstanceOf(LastHttpContent.class);
+
+    if (true) {
```

Comment:
Just use a plain block:
```
{
  Throwable cause ...
  ...
}
```

(Although in this case it seems the extra block doesn't get you much.)